### DISCUSSION TITLE
[Feature Request] Sync duration threshold warning

### DISCUSSION THREAD

**Discussion by matanper** [airbytehq/airbyte#34240] [Connector Ideas and Features]
Airbyte allows to send slack warning when a sync has failed, but I'd like the option to also define a duration threshold for syncs.
For example if I expect a sync to take 20 minutes and it's running for 2 hours, I'd like to get a warning about that

**Comment by marcosmarxm** [comment]
@matanper I think this can be a nice feature. Airbyte can use the average sync time for connections you have some expectations to have always the same data ingested or doing full refresh.

**Reply by matanper** [reply]
I use incremental sync so there are peaks, but I have an upper limit

**Reply by marcosmarxm** [reply]
Definitly using the avg + std deviation can be useful for syncs can finish too fast or are taking too long to finish.

[a comment by an automated account is not shown]
